packages/libp2p/src/libp2p.ts

```diff
@@ -391,6 +392,7 @@ export class Libp2pNode<T extends ServiceMap = Record<string, unknown>> extends
  */
 export async function createLibp2pNode <T extends ServiceMap = Record<string, unknown>> (options: Libp2pOptions<T> = {}): Promise<Libp2pNode<T>> {
   options.peerId ??= await createEd25519PeerId()
+  options.privateKey ??= await unmarshalPrivateKey(options.peerId.privateKey as Uint8Array)
```

Comment:
I think we can skip the cast and make it a bit safer with:

```suggestion
  const peerId = options.peerId ??= await createEd25519PeerId()
  
  if (peerId.privateKey == null) {
    throw new CodeError('peer id was missing private key', 'ERR_MISSING_PRIVATE_KEY')
  }
  
  options.privateKey ??= await unmarshalPrivateKey(peerId.privateKey)
```

This will catch people doing something like this which won't be caught by TypeScript:

```js
const node = await createLibp2p({
  peerId: peerIdFromString('QmFoo')
})
```